src/Chapter3.hs

```diff
@@ -971,6 +1134,26 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+-- | ! Order matters for auto implementation of Enum !
+data Day = Monday | Thursday | Wednesday | Tuesday | Friday | Saturday | Sunday
+    deriving (Show, Eq, Enum, Bounded)
+
+isWeekend :: Day -> Bool
+isWeekend Saturday = True
+isWeekend Sunday   = True
+isWeekend _        = False
+
+nextDay :: Day -> Day
+nextDay day
+    | maxBound  = minBound
```

Comment:
That's a very interesting error, and the compiler is helpless 😛 What you actually wanted to write here is to compare `day` with `maxBound`, and you need to write it like this:

```suggestion
    | day = maxBound  = minBound
```

You can't pattern match on variables, and guards only work with values of type `Bool`. So, in this case, the polymorphic variable `maxBound` is specialized to `Bool`, and its value is `True`. So, currently, your function always returns `minBound`

Reply:
Yes I will make the change but I guess your answer is missing a `=`, isn't it?
```haskell
    | day == maxBound  = minBound
```

Reply:
@JeremyBois Yes, you're correct. One missing `=` in my answer 🙂